# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build gcc-g++-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
cmake -DCMAKE_INSTALL_PREFIX=/ -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/include/glm
install -m 644 glm/common.hpp $DESTDIR/include/glm
install -m 644 glm/exponential.hpp $DESTDIR/include/glm
install -m 644 glm/fwd.hpp $DESTDIR/include/glm
install -m 644 glm/geometric.hpp $DESTDIR/include/glm
install -m 644 glm/glm.hpp $DESTDIR/include/glm
install -m 644 glm/integer.hpp $DESTDIR/include/glm
install -m 644 glm/matrix.hpp $DESTDIR/include/glm
install -m 644 glm/mat2x2.hpp $DESTDIR/include/glm
install -m 644 glm/mat2x3.hpp $DESTDIR/include/glm
install -m 644 glm/mat2x4.hpp $DESTDIR/include/glm
install -m 644 glm/mat3x2.hpp $DESTDIR/include/glm
install -m 644 glm/mat3x3.hpp $DESTDIR/include/glm
install -m 644 glm/mat3x4.hpp $DESTDIR/include/glm
install -m 644 glm/mat4x2.hpp $DESTDIR/include/glm
install -m 644 glm/mat4x3.hpp $DESTDIR/include/glm
install -m 644 glm/mat4x4.hpp $DESTDIR/include/glm
install -m 644 glm/packing.hpp $DESTDIR/include/glm
install -m 644 glm/trigonometric.hpp $DESTDIR/include/glm
install -m 644 glm/vec2.hpp $DESTDIR/include/glm
install -m 644 glm/vec3.hpp $DESTDIR/include/glm
install -m 644 glm/vec4.hpp $DESTDIR/include/glm
install -m 644 glm/vector_relational.hpp $DESTDIR/include/glm
install -d $DESTDIR/include/glm/detail
install -m 644 glm/detail/_fixes.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/_vectorize.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/compute_vector_relational.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_common.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_common_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_exponential.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_exponential_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_geometric.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_geometric_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_integer.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_integer_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_matrix.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_matrix_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_packing.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_packing_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_trigonometric.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_trigonometric_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_vector_relational.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/func_vector_relational_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/qualifier.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/setup.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_float.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_half.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_half.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_int.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x2.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x2.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x3.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x3.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x4.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat2x4.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x2.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x2.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x3.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x3.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x4.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat3x4.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x2.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x2.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x3.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x3.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x4.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x4.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_mat4x4_simd.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec1.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec1.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec2.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec2.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec3.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec3.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec4.hpp $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec4.inl $DESTDIR/include/glm/detail
install -m 644 glm/detail/type_vec4_simd.inl $DESTDIR/include/glm/detail
install -d $DESTDIR/include/glm/ext
install -m 644 glm/ext/vec1.hpp $DESTDIR/include/glm/ext
install -d $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/constants.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/constants.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/epsilon.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/epsilon.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/matrix_inverse.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/matrix_inverse.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/matrix_transform.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/matrix_transform.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/quaternion.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/quaternion.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/quaternion_simd.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/type_ptr.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/type_ptr.inl $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/vec1.hpp $DESTDIR/include/glm/gtc
install -m 644 glm/gtc/vec1.inl $DESTDIR/include/glm/gtc
install -d $DESTDIR/include/glm/simd
install -m 644 glm/simd/common.h $DESTDIR/include/glm/simd
install -m 644 glm/simd/exponential.h $DESTDIR/include/glm/simd
install -m 644 glm/simd/geometric.h $DESTDIR/include/glm/simd
install -m 644 glm/simd/integer.h $DESTDIR/include/glm/simd
install -m 644 glm/simd/matrix.h $DESTDIR/include/glm/simd
install -m 644 glm/simd/platform.h $DESTDIR/include/glm/simd
install -d $DESTDIR/lib/pkgconfig
install -m 644 build/glm.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/glm-build.pc << EOF
Name: glm
Version: 0.9.9.0
Description: Graphics Library Mathematics
Requires: $REQUIRES

devel=\\
/include/glm/common.hpp \\
/include/glm/exponential.hpp \\
/include/glm/fwd.hpp \\
/include/glm/geometric.hpp \\
/include/glm/glm.hpp \\
/include/glm/integer.hpp \\
/include/glm/mat2x2.hpp \\
/include/glm/mat2x3.hpp \\
/include/glm/mat2x4.hpp \\
/include/glm/mat3x2.hpp \\
/include/glm/mat3x3.hpp \\
/include/glm/mat3x4.hpp \\
/include/glm/mat4x2.hpp \\
/include/glm/mat4x3.hpp \\
/include/glm/mat4x4.hpp \\
/include/glm/matrix.hpp \\
/include/glm/packing.hpp \\
/include/glm/trigonometric.hpp \\
/include/glm/vec2.hpp \\
/include/glm/vec3.hpp \\
/include/glm/vec4.hpp \\
/include/glm/vector_relational.hpp \\
/include/glm/detail/_fixes.hpp \\
/include/glm/detail/_vectorize.hpp \\
/include/glm/detail/compute_vector_relational.hpp \\
/include/glm/detail/func_common.inl \\
/include/glm/detail/func_common_simd.inl \\
/include/glm/detail/func_exponential.inl \\
/include/glm/detail/func_exponential_simd.inl \\
/include/glm/detail/func_geometric.inl \\
/include/glm/detail/func_geometric_simd.inl \\
/include/glm/detail/func_integer.inl \\
/include/glm/detail/func_integer_simd.inl \\
/include/glm/detail/func_matrix.inl \\
/include/glm/detail/func_matrix_simd.inl \\
/include/glm/detail/func_packing.inl \\
/include/glm/detail/func_packing_simd.inl \\
/include/glm/detail/func_trigonometric.inl \\
/include/glm/detail/func_trigonometric_simd.inl \\
/include/glm/detail/func_vector_relational.inl \\
/include/glm/detail/func_vector_relational_simd.inl \\
/include/glm/detail/qualifier.hpp \\
/include/glm/detail/setup.hpp \\
/include/glm/detail/type_float.hpp \\
/include/glm/detail/type_half.hpp \\
/include/glm/detail/type_half.inl \\
/include/glm/detail/type_int.hpp \\
/include/glm/detail/type_mat.hpp \\
/include/glm/detail/type_mat2x2.hpp \\
/include/glm/detail/type_mat2x2.inl \\
/include/glm/detail/type_mat2x3.hpp \\
/include/glm/detail/type_mat2x3.inl \\
/include/glm/detail/type_mat2x4.hpp \\
/include/glm/detail/type_mat2x4.inl \\
/include/glm/detail/type_mat3x2.hpp \\
/include/glm/detail/type_mat3x2.inl \\
/include/glm/detail/type_mat3x3.hpp \\
/include/glm/detail/type_mat3x3.inl \\
/include/glm/detail/type_mat3x4.hpp \\
/include/glm/detail/type_mat3x4.inl \\
/include/glm/detail/type_mat4x2.hpp \\
/include/glm/detail/type_mat4x2.inl \\
/include/glm/detail/type_mat4x3.hpp \\
/include/glm/detail/type_mat4x3.inl \\
/include/glm/detail/type_mat4x4.hpp \\
/include/glm/detail/type_mat4x4.inl \\
/include/glm/detail/type_mat4x4_simd.inl \\
/include/glm/detail/type_vec.hpp \\
/include/glm/detail/type_vec1.hpp \\
/include/glm/detail/type_vec1.inl \\
/include/glm/detail/type_vec2.hpp \\
/include/glm/detail/type_vec2.inl \\
/include/glm/detail/type_vec3.hpp \\
/include/glm/detail/type_vec3.inl \\
/include/glm/detail/type_vec4.hpp \\
/include/glm/detail/type_vec4.inl \\
/include/glm/detail/type_vec4_simd.inl \\
/include/glm/ext/vec1.hpp \\
/include/glm/gtc/constants.hpp \\
/include/glm/gtc/constants.inl \\
/include/glm/gtc/epsilon.hpp \\
/include/glm/gtc/epsilon.inl \\
/include/glm/gtc/matrix_inverse.hpp \\
/include/glm/gtc/matrix_inverse.inl \\
/include/glm/gtc/matrix_transform.hpp \\
/include/glm/gtc/matrix_transform.inl \\
/include/glm/gtc/quaternion.hpp \\
/include/glm/gtc/quaternion.inl \\
/include/glm/gtc/quaternion_simd.inl \\
/include/glm/gtc/type_ptr.hpp \\
/include/glm/gtc/type_ptr.inl \\
/include/glm/gtc/vec1.hpp \\
/include/glm/gtc/vec1.inl \\
/include/glm/simd/common.h \\
/include/glm/simd/exponential.h \\
/include/glm/simd/geometric.h \\
/include/glm/simd/integer.h \\
/include/glm/simd/matrix.h \\
/include/glm/simd/platform.h \\
/lib/pkgconfig/glm.pc
EOF

exit
--- glm-0.9.9.0.orig/cmake/GNUInstallDirs.cmake
+++ glm-0.9.9.0/cmake/GNUInstallDirs.cmake
@@ -78,8 +78,10 @@
   # For Debian with multiarch, use 'lib/${CMAKE_LIBRARY_ARCHITECTURE}' if
   # CMAKE_LIBRARY_ARCHITECTURE is set (which contains e.g. "i386-linux-gnu"
   # See http://wiki.debian.org/Multiarch
+  execute_process(COMMAND gcc -print-multi-os-directory OUTPUT_VARIABLE MULTIOSDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
   if((CMAKE_SYSTEM_NAME MATCHES "Linux|kFreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GNU")
-      AND NOT CMAKE_CROSSCOMPILING)
+      AND NOT CMAKE_CROSSCOMPILING
+      AND NOT ("${CMAKE_SIZEOF_VOID_P}" EQUAL "8" AND "${MULTIOSDIR}" STREQUAL "../lib"))
     if (EXISTS "/etc/debian_version") # is this a debian system ?
        if(CMAKE_LIBRARY_ARCHITECTURE)
          set(_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
